### PR TITLE
Restore base quality filter code that got reverted in #4895

### PR DESCRIPTION
**Pull request by takutosato**
This unintended change crept in with #4895.

@meganshand @davidbenjamin

### CODE REVIEW

**Comment by takutosato** [comment]
@meganshand please review!

**Comment by takutosato** [comment]
@meganshand I added a very specific test to check for this bug that I introduced. I confirmed that the test fails before this change and passes after.

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/readorientation/CollectF1R2CountsIntegrationTest.java [hunk 1]
```diff
@@ -201,8 +201,6 @@ private File createSyntheticSam(final int refDepth, final int altDepth) throws I
 
         // create a sample list
         final int chromosomeIndex = 0;
-        final String sampleName = "samthreetree";
```
Comment (meganshand) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/readorientation/CollectF1R2CountsIntegrationTest.java]: Did you mean to make these changes? They look unrelated to the other test.

**Comment by sbamin** [comment]
HI @takutosato,

https://github.com/broadinstitute/gatk/blob/9df7b59b611d8051cd7b1fc2eac08617e25e8a22/src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2FilteringEngine.java#L115-L117

Looks like this could have impacted one or more of mutect2 somatic filtering ([doc-11136](https://software.broadinstitute.org/gatk/documentation/article?id=11136)). I believe bug was related to an argument, `--min-median-base-quality`, i.e., it was either ignoring this argument or (hope) taking a default value of 20 in Mutect2 and FilterMutectCalls tools.

* Is that the correct interpretation?
* Would this mandate running mutect2 from scratch or only FilterMutectCalls and subsequent filtration steps? 

I am looking through our mutect2 filtered calls and filtering rules applied. Among PASS calls, so far, I don't see false positive with respect to `FORMAT/MBQ < 20`. Among filtered out calls, I don't have calls failing due to single filter but I need to look more into raw calls. Anyways, it would be of help to get your comments on possible impact of this bug in gatk version < 4.0.9.0 (I used 4.0.8.1).
